charms/worker/k8s/src/charm.py

```diff
@@ -1064,7 +1068,10 @@ def _apply_extra_args(self):
             status.add(ops.MaintenanceStatus("Ensuring Kubernetes Extra Args"))
             file_args_config = config.arg_files.FileArgsConfig()
             node_ips = self._get_node_ips()
-            config.extra_args.craft(self.config, file_args_config, cluster_name, node_ips)
+            datastore = self.bootstrap.config.datastore if self.is_control_plane else None
```

Comment:
You had something that's overkill.  `self.bootstrap.config.datastore` is already defined as an `Optional[str]` and will be None on the worker units.

```suggestion
            datastore = self.bootstrap.config.datastore
```